### PR TITLE
Separate RMC sanity check function

### LINKED ISSUE #200: Add an `rmc_sanity_check` function for that flavor of assert
For vtable sanity checks, we want to add assertions that are essentially cheap checks that our codegen reasoning is correct. It would be nice to have a separate path for these that add the issue-filing URL to assertions and hard fail at that point, similar to `codegen_unimplemented`.

### PR DESCRIPTION
**Pull request by avanhatt**
### Description of changes: 

Add a clearly-named function for CBMC-time assertions that indicate codegen failures that can only be detected then. Similar to `codegen_unimplemented`, refer users to file an issue when they see these assertions fail.

### Resolved issues:

Resolves #200

### Call-outs:

Right now, the vtable size check seems to be the only place we use this, but @danielsn and I agreed we might want to add more in the future.

### Testing:

* How is this change tested?

Existing tests.

* Is this a refactor change?

A very small one, yes.

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: compiler/rustc_codegen_llvm/src/gotoc/metadata.rs [hunk 1]
```diff
@@ -394,6 +394,37 @@ impl<'tcx> GotocCtx<'tcx> {
...
+            vec![
+                // Assert our expected true expression.
+                Stmt::assert(expect_true.clone(), &assert_msg, loc.clone()),
+                // If it fails, assume false to block any further exploration of this path.
+                Stmt::if_then_else(
+                    expect_true.not(),
+                    Stmt::assume(Expr::bool_false(), loc.clone()),
+                    None,
+                    loc.clone(),
+                ),
```
Comment (danielsn) [inline compiler/rustc_codegen_llvm/src/gotoc/metadata.rs:423]: Why not just `assume(expect_true)`

File: compiler/rustc_codegen_llvm/src/gotoc/metadata.rs [hunk 2]
```diff
@@ -394,6 +394,37 @@ impl<'tcx> GotocCtx<'tcx> {
 
         Expr::statement_expression(body, t).with_location(loc)
     }
+
+    /// RMC-specific function to sanity check expected components of in code
+    /// generation. If users see these assertions fail, something in our
+    /// translation to Gotoc has gone wrong, and we want them to file an issue.
+    pub fn codegen_sanity_check(
```
Comment (danielsn) [inline compiler/rustc_codegen_llvm/src/gotoc/metadata.rs:401]: I don't think this uses `self` at all.  Make it a compound constructor on `goto_program::Stmt`

**Review by danielsn — CHANGES_REQUESTED** [review]

File: compiler/rustc_codegen_llvm/src/gotoc/metadata.rs (hunk 1 above)
Comment (avanhatt) [reply compiler/rustc_codegen_llvm/src/gotoc/metadata.rs:423]: Ahhh yes much cleaner

File: compiler/rustc_codegen_llvm/src/gotoc/cbmc/goto_program/stmt.rs [hunk 3]
```diff
@@ -187,6 +187,28 @@ impl Stmt {
         Stmt::assert(Expr::bool_false(), msg, loc)
     }
 
+    /// RMC-specific function to sanity check expected components of in code
```
Comment (danielsn) [inline compiler/rustc_codegen_llvm/src/gotoc/cbmc/goto_program/stmt.rs:190]: Lets make this generic, to make this easier to split into its own package.

File: compiler/rustc_codegen_llvm/src/gotoc/cbmc/goto_program/stmt.rs [hunk 4]
```diff
@@ -187,6 +187,28 @@ impl Stmt {
...
+    /// generation. If users see these assertions fail, something in our
+    /// translation to Gotoc has gone wrong, and we want them to file an issue.
+    pub fn assert_sanity_check(expect_true: Expr, message: &str, loc: Location) -> Stmt {
+        let url = "https://github.com/model-checking/rmc/issues/new?template=bug_report.md";
```
Comment (danielsn) [inline compiler/rustc_codegen_llvm/src/gotoc/cbmc/goto_program/stmt.rs:194]: take this as a paramater.

**Review by danielsn — CHANGES_REQUESTED** [review]

File: compiler/rustc_codegen_llvm/src/gotoc/cbmc/goto_program/stmt.rs (hunk 3 above)
Comment (avanhatt) [reply compiler/rustc_codegen_llvm/src/gotoc/cbmc/goto_program/stmt.rs]: Done!

File: compiler/rustc_codegen_llvm/src/gotoc/cbmc/goto_program/stmt.rs (hunk 4 above)
Comment (avanhatt) [reply compiler/rustc_codegen_llvm/src/gotoc/cbmc/goto_program/stmt.rs]: Moved

**Review by danielsn — APPROVED** [review]

**Pull request merged**